src/Middleware/UserPageDecorationMiddleware.php

```diff
@@ -151,6 +151,14 @@ function ($a, $b) {
             );
         }
 
+        // This should be moved out of the middleware and into the modules that need it,
```

Comment:
Why are you moving this? I don't see why this PR should touch any middleware

Reply:
Hi @driusan the new placement here, resolves the data not being refreshed and specifically when it's being deleted. Otherwise the browser had to be refreshed (unnecessary) for the data that was deleted to be displayed correctly as gone.

Reply:
I don't think that should be within the scope of this PR and the comment leads me to believe that it's likely to introduce bugs in other modules or features. A simpler (and less risky) solution is to check if "ClearInstrument" it's a POST request with a 'ClearInstrument' element set and include a Refresh header in the response or a "205 Reset Content" response code (which might need to be handled properly in js)